lib/sshutils/server.go

```diff
@@ -486,10 +485,21 @@ func (s *Server) HandleConnection(conn net.Conn) {
 		defer s.ingressReporter.ConnectionClosed(s.ingressService, conn)
 	}
 
-	cfg := &s.cfg
+	hostSigners := s.getHostSigners()
+	if err := s.validateHostSigners(hostSigners); err != nil {
+		s.log.
+			WithError(err).
+			WithField("remote_addr", conn.RemoteAddr()).
+			Error("Error during server setup for a new SSH connection (this is a bug).")
+		conn.Close()
+		return
+	}
+
+	cfg := s.cfg
+	for _, signer := range hostSigners {
+		cfg.AddHostKey(signer)
+	}
```

Comment:
The validation in `(*Server).checkArguments` uses `validateHostSigners` against no specific `ssh.ServerConfig` so it would require a way to _optionally_ add the validated keys to the config; what would be the point, anyway?

Also, we're likely only going to have one signer there anyway (unless we start doing multi-algorithm host credentials, but that's kind of weird).